### ISSUE TITLE
The right way to build a workspace [kubernetes]

### ISSUE THREAD

**Issue by solesensei** [loft-sh/devpod#920]
Hey there! 👋

I'm currently setting up workspace builds.

```jsonc
  // .devcontainer.json
  "build": {
    "context": "..",
    "dockerfile": "../Dockerfile"
  },
```

Previously, I was able to use `BUILD_REPOSITORY` and `BUILDKIT` options to build (on kubernetes) and push an image to some image registry. This way, I could avoid rebuilding my workspace each time I launched it. However, it seems that now the only method available is to use the `devpod build` command and prebuild workspaces, but this approach is limited to the `docker` driver and my laptop, not the kubernetes.

So, my main question is: What's the current recommended approach for building and saving workspaces and is it possible to do so on Kubernetes?

**Comment by pascalbreuninger** [comment]
Hi @solesensei, we're now building the workspace directly in the pod instead of on the users machine like before, which is why there are no BuildKit options anymore. 
The new flow for this is using prebuilds, the piece you're missing in your setup is only step 1:
1. Specify a prebuild customization in your `devcontainer.json`:
```
...
 "customizations": {
    "devpod": {
      "prebuildRepository": "ghcr.io/my-org/my-repo"
    }
  }```
 2. Build the prebuild image locally or in a CI Pipeline `devpod build github.com/my-org/my-repo --repository ghcr.io/my-org/my-repo`
 3. Create the workspace using the kubernetes provider. If there is a prebuild match it'll prefer that over building from source

**Comment by solesensei** [comment]
Hi @pascalbreuninger, thanks for the clarification!

So if I understood correctly the point 2. there is no way to build&push on Kubernetes itself.

What kind of CI pipeline can you recommend? Do you mean Github Actions with some devpod action?

**Comment by solesensei** [comment]
As far as I remember buildkit can build your image on a temporary buildkit pod and then push it to the image registry. Are there any specific reasons why you have stopped supporting it?

**Comment by pascalbreuninger** [comment]
> As far as I remember buildkit can build your image on a temporary buildkit pod and then push it to the image registry. Are there any specific reasons why you have stopped supporting it?

Yes, we wanted to get rid of the requirement to have a registry configured. Now we use [Kaniko](https://github.com/GoogleContainerTools/kaniko) to build and start in the same pod, without the necessity to push the image anywhere 

As for 
> What kind of CI pipeline can you recommend? Do you mean Github Actions with some devpod action?

We were thinking about writing a github action to make that easier, generally any CI solution that supports docker should suffice. The basic steps there would be, for example triggered by changes to the `.devcontainer` folder:
1. Login to docker
2. Download DevPod CLI 
3. `devpod provider add docker` 
4. `devpod build github.com/my-org/my-repo --repository ghcr.io/my-org/my-repo --provider docker`
